signal/signals.go

```diff
@@ -40,8 +41,12 @@ func send(typ string, event interface{}) {
 	data, err := json.Marshal(&signal)
 	if err != nil {
 		logger.Error("Marshalling signal envelope", "error", err)
+		return
 	}
-	C.StatusServiceSignalEvent(C.CString(string(data)))
+
+	str := C.CString(string(data))
+	C.StatusServiceSignalEvent(str)
+	C.free(unsafe.Pointer(str))
```

Comment:
Have you verified that we don't cause any invalid memory reads on Android and iOS? For example, in Android, `C.StatusServiceSignalEvent` we do `if (javaJsonEvent != NULL) (*env)->DeleteLocalRef(env, javaJsonEvent);`.

Reply:
@adambabik no, I haven't. But as I understand, CCstring always make a copy and responsibility to free its memory is on the caller.

`javaJsonEvent` is a separate chunk of memory so it's ok that it is deleted there:
```c
    jstring javaJsonEvent = NULL;
    if (jsonEvent != NULL) {
        javaJsonEvent = (*env)->NewStringUTF(env, jsonEvent);
    }
```